### ISSUE TITLE
Emails not being sent on clean install

### ISSUE THREAD

**Issue by karatekaneen** [getsentry/onpremise#807]
## Version Information

Version: 20.12.1

## TL;DR

Since I can not send emails, are there any workaround to register users? This will solve most of my issues at the moment.

## Description

When doing a clean install on the latest release emails are not being sent properly.
I followed the guide that said "All you need to do is to set a valid address for mail.from setting in config.yml" and left everything else as default. So my email part of `config.yml` looks like this:

```
###############
# Mail Server #
###############

# mail.backend: 'smtp'  # Use dummy if you want to disable email entirely
mail.host: 'smtp'
# mail.port: 25
# mail.username: ''
# mail.password: ''
# mail.use-tls: false
# The email address to send on behalf of
mail.from: 'noreply@mycompany.se'
```

## Steps to Reproduce

1. Change the config file
2. Install and run the containers
3. Invite member by email
4. Not receive email

### SMTP - Gmail

Note that the send is marked as a success both in the logs and in the UI with `09:48:46 [INFO] sentry.mail: mail.sent`

![Skärmavbild 2021-01-13 kl  10 56 13](https://user-images.githubusercontent.com/41196840/104436622-1e7a9880-558e-11eb-8c3e-88900187fb28.png)

```
smtp_1 |   271 LOG: MAIN
smtp_1 |   271   <= noreply@ mycompany.se H=sentry_onpremise_worker_1.sentry_onpremise_default (d74daf83b17e) [172.22.0.21] P=esmtp S=5146 id=20210113094845.22.278@ mycompany.se
smtp_1 |   271 LOG: smtp_connection MAIN
smtp_1 |   271   SMTP connection from sentry_onpremise_worker_1.sentry_onpremise_default (d74daf83b17e) [172.22.0.21] closed by QUIT
smtp_1 |   272 Exim version 4.92 uid=101 gid=101 pid=272 D=80001
worker_1 | 09:48:46 [INFO] sentry.mail: mail.sent (message_id='<20210113094845.22.278@ mycompany.se>' size=4896)
smtp_1 | Berkeley DB: Berkeley DB 5.3.28: (September  9, 2013)
smtp_1 | Support for: crypteq iconv() IPv6 GnuTLS move_frozen_messages DANE DKIM DNSSEC Event OCSP PRDR SOCKS TCP_Fast_Open
smtp_1 | Lookups (built-in): lsearch wildlsearch nwildlsearch iplsearch cdb dbm dbmjz dbmnz dnsdb dsearch nis nis0 passwd
smtp_1 | Authenticators: cram_md5 plaintext
smtp_1 | Routers: accept dnslookup ipliteral manualroute queryprogram redirect
smtp_1 | Transports: appendfile/maildir/mailstore autoreply lmtp pipe smtp
smtp_1 | Fixed never_users: 0
smtp_1 | Configure owner: 0:0
smtp_1 | Size of off_t: 8
smtp_1 |   272 delivering 1kzMot-00004N-Kq
smtp_1 |   272 R: dnslookup for my_full_name@gmail.com
smtp_1 |   273 T: remote_smtp for my_full_name@gmail.com
smtp_1 |   273 Connecting to gmail-smtp-in.l.google.com [2a00:1450:400c:c00::1b]:25 ... failed: Cannot assign requested address
smtp_1 |   273 LOG: MAIN
smtp_1 |   273   H=gmail-smtp-in.l.google.com [2a00:1450:400c:c00::1b] Cannot assign requested address
smtp_1 |   273 Connecting to gmail-smtp-in.l.google.com [173.194.76.26]:25 ... failed: Connection timed out (timeout=5m)
smtp_1 |   273 LOG: MAIN
smtp_1 |   273   H=gmail-smtp-in.l.google.com [173.194.76.26] Connection timed out
smtp_1 |   273 Connecting to alt1.gmail-smtp-in.l.google.com [2a00:1450:4010:c03::1a]:25 ... failed: Cannot assign requested address
smtp_1 |   273 LOG: MAIN
smtp_1 |   273   H=alt1.gmail-smtp-in.l.google.com [2a00:1450:4010:c03::1a] Cannot assign requested address
smtp_1 |   273 Connecting to alt1.gmail-smtp-in.l.google.com [209.85.233.26]:25 ... failed: Connection timed out (timeout=5m)
smtp_1 |   273 LOG: MAIN
smtp_1 |   273   H=alt1.gmail-smtp-in.l.google.com [209.85.233.26] Connection timed out
smtp_1 |   273 Connecting to alt2.gmail-smtp-in.l.google.com [2404:6800:4003:c06::1b]:25 ... failed: Cannot assign requested address
smtp_1 |   273 LOG: MAIN
smtp_1 |   273   H=alt2.gmail-smtp-in.l.google.com [2404:6800:4003:c06::1b] Cannot assign requested address
smtp_1 |   273 Connecting to alt3.gmail-smtp-in.l.google.com [2404:6800:4008:c00::1a]:25 ... failed: Cannot assign requested address
smtp_1 |   273 LOG: MAIN
smtp_1 |   273   H=alt3.gmail-smtp-in.l.google.com [2404:6800:4008:c00::1a] Cannot assign requested address
smtp_1 |   273 Connecting to alt4.gmail-smtp-in.l.google.com [2607:f8b0:400e:c04::1a]:25 ... failed: Cannot assign requested address
smtp_1 |   273 LOG: MAIN
smtp_1 |   273   H=alt4.gmail-smtp-in.l.google.com [2607:f8b0:400e:c04::1a] Cannot assign requested address
smtp_1 |   272 LOG: MAIN
smtp_1 |   272   == my_full_name@gmail.com R=dnslookup T=remote_smtp defer (99): Cannot assign requested address
```

### SMTP - Hotmail

```
smtp_1 | Berkeley DB: Berkeley DB 5.3.28: (September  9, 2013)
smtp_1 | Support for: crypteq iconv() IPv6 GnuTLS move_frozen_messages DANE DKIM DNSSEC Event OCSP PRDR SOCKS TCP_Fast_Open
smtp_1 | Lookups (built-in): lsearch wildlsearch nwildlsearch iplsearch cdb dbm dbmjz dbmnz dnsdb dsearch nis nis0 passwd
smtp_1 | Authenticators: cram_md5 plaintext
smtp_1 | Routers: accept dnslookup ipliteral manualroute queryprogram redirect
smtp_1 | Transports: appendfile/maildir/mailstore autoreply lmtp pipe smtp
smtp_1 | Fixed never_users: 0
smtp_1 | Configure owner: 0:0
smtp_1 | Size of off_t: 8
smtp_1 |   275 delivering 1kzN51-00004Q-Gu
smtp_1 |   275 R: dnslookup for  my_name_@hotmail.com
smtp_1 |   276 T: remote_smtp for  my_name_@hotmail.com
smtp_1 |   276 Connecting to hotmail-com.olc.protection.outlook.com [104.47.58.33]:25 ... failed: Connection timed out (timeout=5m)
smtp_1 |   276 LOG: MAIN
smtp_1 |   276   H=hotmail-com.olc.protection.outlook.com [104.47.58.33] Connection timed out
smtp_1 |   276 Connecting to hotmail-com.olc.protection.outlook.com [104.47.55.33]:25 ... failed: Connection timed out (timeout=5m)
smtp_1 |   276 LOG: MAIN
smtp_1 |   276   H=hotmail-com.olc.protection.outlook.com [104.47.55.33] Connection timed out
smtp_1 |   275 LOG: MAIN
smtp_1 |   275   ==  my_name_@hotmail.com R=dnslookup T=remote_smtp defer (110): Connection timed out
```

### Ping

Just to be sure that there wasnt any connection issues from the VM I pinged the URLs tried:

```
robin@sentry:~/onpremise-20.12.1$ ping gmail-smtp-in.l.google.com
PING gmail-smtp-in.l.google.com (74.125.133.26) 56(84) bytes of data.
64 bytes from wo-in-f26.1e100.net (74.125.133.26): icmp_seq=1 ttl=109 time=8.25 ms
```

_edit: added more logs + screenshot of ui_

**Comment by karatekaneen** [comment]
Update: switched over to using `smtp-relay.gmail.com` instead and it now works, so my guess is that there is something wonky in the default settings.

Thanks for the amazing work you do.

**Comment by chadwhitacre** [comment]
Thanks for the solid report and for following up @karatekaneen, glad you got it sorted. Since the default `smtp` backend we ship with is really just a toy/dev server, switching to a real server sounds like the right move.

**Comment by BYK** [comment]
@karatekaneen - just googled the resolved hotmail domain as it looked fishy to me. I think your server is being blackholed for SMTP due to potential spam: https://answers.microsoft.com/en-us/outlook_com/forum/osettings-oemailset/how-do-l-remove-our-server-ip-address-from-the/699830e9-e2fc-4699-8183-539c152bd388
